client/src/pages/board/KanbanBoard.jsx

```diff
@@ -0,0 +1,297 @@
+    width: '30vh',
+    overflow: 'hidden',
+    '&#leftNav': {
+      '& .addColumnContent': {
+        minHeight: '500px',
+        '& a': {
+          width: '2%',
+          padding: '0 30px 0 0',
+          left: '-30px',
+          minHeight: '500px',
+          position: 'fixed',
+          textDecoration: 'none',
+          height: 'inherit',
+          display: 'flex',
+          alignItems: 'center',
+          '& button': {
+            marginRight: '10px',
+            backgroundColor: 'lightgrey',
+            padding: '0px',
+            color: 'grey',
+          },
+          '&:hover': {
+            padding: '0 0 0 20px',
+            left: 0,
+            position: 'relative',
+            width: 'fit-content',
+            backgroundColor: 'lightgrey',
+            transition: '0.05s',
+          },
+        },
+      },
+    },
+    '&#rightNav': {
+      display: 'flex',
+      justifyContent: 'flex-end',
+      '& .addColumnContent': {
+        minHeight: '500px',
+        '& a': {
+          position: 'fixed',
+          transition: '0.1s',
+          textDecoration: 'none',
+          height: 'inherit',
+          display: 'flex',
+          alignItems: 'center',
+          width: '2%',
+          padding: '0 0 0 30px',
+          right: '-30px',
+          minHeight: '500px',
+          '& button': {
+            marginLeft: '10px',
+            backgroundColor: 'lightgrey',
+            color: 'grey',
+            padding: '0px',
+          },
+          '&:hover': {
+            position: 'relative',
+            width: 'fit-content',
+            backgroundColor: 'lightgrey',
+            right: 0,
+            padding: '0 10px 0 0',
+          },
+        },
+      },
+    },
+  },
+}));
+
+// performance optimization. prevents re-render when components are dragged all over w/memo
+const InnerList = memo((props) => {
+  const { column, taskMap, index, setUpdate = { setUpdate } } = props;
+  const tasks = column.taskIds.map((taskId) => taskMap[taskId]);
+  return <Column column={column} tasks={tasks} index={index} setUpdate={setUpdate} />;
+});
+
+export default function KanbanBoard() {
+  const [data, setData] = useState({
+    id: '',
+    tasks: {},
+    columns: {},
+    columnOrder: []
+  });
+
+  const [update, setUpdate] = useState(true)
+
+  const convertAPIData = async () => {
+    const boardData = await getBoard()
+    
+    const data = await boardData.data
+    
+    const newData = {
+      id: data._id,
+      tasks: {},
+      columns: {},
+      columnOrder: [],
+    }
+    await data.columns.map(column => {
+      newData.columns[column._id] = {
+        id: column._id,
+        title: column.name,
+        taskIds: column.cards.map(card => card._id)
+      }
+      column.cards.map(card => {
+        newData.tasks[card._id] = {
+          id: card._id,
+          content: card.name
+        }
+      })
+      newData.columnOrder.push(column._id)
+    })
+    setData(newData)
+  }
```

Comment:
I'm going to refactor this into the getBoard() api call and send back the data as the react dnd state would need. I also agree that this is something to keep out of the component.

Reply:
There is no need to modify the backend. We should make any necessary translations between backend responses and frontend spec in the frontend api layer. E.g. this would look something like:

    // NOT ACTUAL CODE
    const myApi (args) => {
        const {data}= axios.getWhatever()

         // take only whatever is useful for frontend
        const {_id : id, myBackendProp : renameToWhatFrontendExpects, ... } = data
        return ApiResult.withData({ id, frontendProp1, ...});
    };